presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -6888,4 +6888,30 @@ public void testAlterTableWithStructDataTypeColumns()
         assertUpdate("DROP TABLE testReadSchema1.testReadStruct6");
         assertUpdate("DROP SCHEMA testReadSchema1");
     }
+
+    @Test
+    public void testUpdateWithStructDataTypeColumns()
+    {
+        assertUpdate("CREATE SCHEMA testReadSchema2");
+        assertUpdate("CREATE TABLE testReadSchema2.testReadStruct7 (orderkey array(varchar), orderstatus array(row(big int)), lables map(varchar, array(int)), orderstatus1 row(big2 int))  WITH (format='orc',transactional=true)");
+        assertUpdate("INSERT INTO testReadSchema2.testReadStruct7 VALUES (ARRAY ['YOUNG', 'FASION', 'STYLE'], ARRAY[row(111), row(222)], MAP(ARRAY ['type', 'grand'], ARRAY [ARRAY[1, 2], ARRAY[1,2]]), row(111))", 1);
+        assertEquals(computeActual("SELECT * FROM testReadSchema2.testReadStruct7").getRowCount(), 1);
+        assertUpdate("UPDATE testReadSchema2.testReadStruct7 set orderkey = ARRAY['UPDATE','UPDATE','UPDATE']", 1);
+        assertEquals(computeActual("SELECT * FROM testReadSchema2.testReadStruct7").getRowCount(), 1);
+        assertUpdate("DROP TABLE testReadSchema2.testReadStruct7");
+        assertUpdate("DROP SCHEMA testReadSchema2");
+    }
+
+    @Test
+    public void testReadSingleColumnStructDataTypeColumns()
+    {
+        assertUpdate("CREATE SCHEMA testReadSchema3");
+        assertUpdate("CREATE TABLE testReadSchema3.testReadStruct8 (orderkey array(varchar), orderstatus array(row(big int)), lables map(varchar, array(int)), orderstatus1 row(big2 int))  WITH (format='orc',transactional=true)");
```

Comment:
add test case with native types and complex types alternating types.
add queries for : -
a) each Col name separately
b) combinations of colms with difference of 2 or more colindex,

Reply:
added test case